collections.js

```diff
@@ -118,8 +118,79 @@ LinkedList.prototype.find = function(value) {
     node = node.next;
   }
   return null;
+};
+
+//------------------------------------------------------------------------------
+// Simple Queue - a trivial queue implementation used to verify tests.
+// The head of the queue is at the beginning of the backing array.
+
+function SimpleQueue() {
```

Comment:
It's weird to have this exposed as part of the API, since it's not useful as it stands. You don't appear to use it in your implementation of Queue either.